include/util/static_rtree.hpp

```diff
@@ -49,71 +49,150 @@ namespace osrm
+     *            \   /       /
+     *             \ /       /
+     *              U       V
+     *               \     /
+     *                \   /
+     *                 \ /
+     *                  W
+     *
+     * Step 1 - objects 01234567... are sorted by Hilbert code (these are the line
+     *          segments of the OSM roads)
+     * Step 2 - we grab LEAF_NODE_SIZE of them at a time and create TreeNode A with a
+     *          bounding-box that surrounds the first LEAF_NODE_SIZE objects
+     * Step 2a- continue grabbing LEAF_NODE_SIZE objects, creating TreeNodes B,C,D,E...J
+     *          until we run out of objects.  The last TreeNode J may not have
+     *          LEAF_NODE_SIZE entries.  Our math later on caters for this.
+     * Step 3 - Now start grabbing nodes from A..J in groups of BRANCHING_FACTOR,
+     *          and create K..O with bounding boxes surrounding the groups of
+     *          BRANCHING_FACTOR.  Again, O, the last entry, may have fewer than
+     *          BRANCHING_FACTOR entries.
+     * Step 3a- Repeat this process for each level, until you only create 1 TreeNode
+     *          to contain its children (in this case, W).
+     *
+     * As we create TreeNodes, we append them to the m_search_tree vector.
+     *
+     * After this part of the building process, m_search_tree will contain TreeNode
+     * objects in this order:
+     *
+     * ABCDEFGHIJ KLMNO PQR UV W
+     * 10         5     3   2  1  <- number of nodes in the level
+     *
+     * In order to make our math easy later on, we reverse the whole array,
+     * then reverse the nodes within each level:
+     *
+     *   Reversed:        W VU RQP ONMKL JIHGFEDCBA
+     *   Levels reversed: W UV PQR KLMNO ABCDEFGHIJ
+     *
+     * We also now have the following information:
+     *
+     *   level sizes = {1,2,3,5,10}
+     *
+     * and we can calculate the array position the nodes for each level
+     * start (based on the sum of the previous level sizes):
+     *
+     *   level starts = {0,1,3,6,11}
+     *
+     * Now, some basic math can be used to navigate around the tree.  See
+     * the body of the `child_indexes` function for the details.
+     *
+     ***********************************************/
     template <typename T> using Vector = ViewOrVector<T, Ownership>;
 
   public:
     using Rectangle = RectangleInt2D;
     using EdgeData = EdgeDataT;
     using CoordinateList = Vector<util::Coordinate>;
 
-    static_assert(LEAF_PAGE_SIZE >= sizeof(uint32_t) + sizeof(EdgeDataT), "page size is too small");
+    static_assert(LEAF_PAGE_SIZE >= sizeof(EdgeDataT), "page size is too small");
     static_assert(((LEAF_PAGE_SIZE - 1) & LEAF_PAGE_SIZE) == 0, "page size is not a power of 2");
-    static constexpr std::uint32_t LEAF_NODE_SIZE =
-        (LEAF_PAGE_SIZE - sizeof(uint32_t) - sizeof(Rectangle)) / sizeof(EdgeDataT);
+    static constexpr std::uint32_t LEAF_NODE_SIZE = (LEAF_PAGE_SIZE / sizeof(EdgeDataT));
```

Comment:
`LEAF_NODE_SIZE` can be a template parameter similar to `BRANCHING_FACTOR` without computing from `LEAF_PAGE_SIZE`